lib/ecto/changeset.ex

```diff
@@ -106,17 +111,19 @@ defmodule Ecto.Changeset do
 
   def change(%Changeset{changes: changes} = changeset, new_changes)
       when is_map(new_changes) do
-    %{changeset | changes: get_changed(changeset.model, changes, new_changes)}
+    changeset = ensure_has_types(changeset)
```

Comment:
Is this a smell? Why was a changeset created without types? Maybe we should be raising on put_change if there is no type instead of "ensuring"?